crates/pop-parachains/src/call/metadata/mod.rs

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use params::Param;
+use scale_value::stringify::custom_parsers;
+use std::fmt::{Display, Formatter};
+use subxt::{dynamic::Value, Metadata, OnlineClient, SubstrateConfig};
+
+pub mod action;
+pub mod params;
+
+/// Represents a pallet in the blockchain, including its extrinsics.
+#[derive(Clone, PartialEq, Eq)]
+pub struct Pallet {
+	/// The name of the pallet.
+	pub name: String,
+	/// The documentation of the pallet.
+	pub docs: String,
+	/// The extrinsics of the pallet.
+	pub extrinsics: Vec<Extrinsic>,
+}
+
+impl Display for Pallet {
+	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+		write!(f, "{}", self.name)
+	}
+}
+
+/// Represents an extrinsic in a pallet.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Extrinsic {
+	/// The name of the extrinsic.
+	pub name: String,
+	/// The documentation of the extrinsic.
+	pub docs: String,
+	/// The parameters of the extrinsic.
+	pub params: Vec<Param>,
+	/// Whether this extrinsic is supported (no recursive or unsupported types like `RuntimeCall`).
+	pub is_supported: bool,
+}
+
+impl Display for Extrinsic {
+	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+		write!(f, "{}", self.name)
+	}
+}
+
+/// Parses the chain metadata to extract information about pallets and their extrinsics with its
+/// parameters.
+///
+/// # Arguments
+/// * `client`: Reference to an `OnlineClient` connected to the chain.
```

Comment:
leave out reference

Reply:
You are right, improved here https://github.com/r0gue-io/pop-cli/pull/316/commits/e70bc7a6dbefd8e37165f102654d012605c51a37